go/runtime/registry/registry.go

```diff
@@ -236,6 +256,34 @@ func (r *runtimeRegistry) Cleanup() {
 	}
 }
 
+func (r *runtimeRegistry) FinishInitialization(ctx context.Context) (rerr error) {
+	r.RLock()
+	defer r.RUnlock()
+
+	for _, rt := range r.runtimes {
+		if rt.storage == nil {
+			storageBackend, err := client.New(ctx, rt.id, r.identity, rt.consensus.Scheduler(), rt.consensus.Registry(), rt)
+			if err != nil {
+				return fmt.Errorf("runtime/registry: cannot create storage for runtime %s: %w", rt.id, err)
+			}
+			if err = rt.tagIndexer.Start(storageBackend); err != nil {
+				storageBackend.Cleanup()
+				return fmt.Errorf("runtime/registry: cannot start tag indexer for runtime %s: %w", rt.id, err)
+			}
+			rt.storage = storageBackend
+			rt.finished = true
+			defer func() {
+				if rerr != nil {
+					rt.tagIndexer.Stop()
+					storageBackend.Cleanup()
+				}
+			}()
+		}
```

Comment:
Could this part be moved into the `runtime`, so you would call something like `rt.finishInitialization()` here?